tests/subsys/logging/log_core_additional/prj.conf

```diff
@@ -9,3 +9,5 @@ CONFIG_LOG_PROCESS_TRIGGER_THRESHOLD=0
 CONFIG_THREAD_MONITOR=y
 CONFIG_LOG_MODE_DEFERRED=y
 CONFIG_MEM_SLAB_TRACE_MAX_UTILIZATION=y
+CONFIG_ASSERT=n
+CONFIG_SPIN_VALIDATE=n
```

Comment:
So... the primary hypothesis when a change to remove validation code is claimed to fix a bug has to be that the code in question is triggering those validation cases legitimately and that we should fix it. :)

It's admittedly true that SPIN_VALIDATE can produce some hard-to-diagnose failures (I had a few days of fun after I added that trying to figure out where a "recursive spinlock" was being taken when the real problem was a errant case in z_swap() that was straight up breaking locks silently).

But we should figure this out.